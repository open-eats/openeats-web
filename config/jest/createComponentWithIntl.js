import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

// See: https://github.com/yahoo/react-intl/wiki/Testing-with-React-Intl#jest

const createComponentWithIntl = (children, props = { locale: 'en' }) => {
  return renderer.create(
    <IntlProvider {...props}>
      {children}
    </IntlProvider>
  );
};

export default createComponentWithIntl;