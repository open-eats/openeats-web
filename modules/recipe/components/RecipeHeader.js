import React from 'react'
import PropTypes from 'prop-types'

import Ratings from '../../rating/components/Ratings'

const RecipeHeader = ({ photo, title, rating, addToMenu }) => {
  if (photo) {
    return (
      <div className="panel-heading hero-image" style={{backgroundImage: "url(" + photo + ")"}}>
        <div className="row title">
          <div className="col-xs-12">
            <h3>{ title }</h3>
            <Ratings stars={ rating }/>
          </div>
        </div>
        <div className="row options print-hidden">
          <div className="col-xs-12">
            <button className="btn btn-primary" onClick={addToMenu}>
              <span className="glyphicon glyphicon-calendar" aria-hidden="true"/>
            </button>
            <button className="btn btn-primary" onClick={ window.print }>
              <span className="glyphicon glyphicon-print" aria-hidden="true"/>
            </button>
          </div>
        </div>
      </div>
    );
  }

  return (
    <div className="panel-heading">
      <div className="row">
        <div className="col-xs-12">
          <h3>{ title }</h3>
          <Ratings stars={ rating }/>
        </div>
      </div>
      <div className="row options print-hidden">
        <div className="col-xs-12">
          <button className="btn btn-primary" onClick={addToMenu}>
            <span className="glyphicon glyphicon-calendar" aria-hidden="true"/>
          </button>
          <button className="btn btn-primary" onClick={ window.print }>
            <span className="glyphicon glyphicon-print" aria-hidden="true"/>
          </button>
        </div>
      </div>
    </div>
  );
};

RecipeHeader.PropTypes = {
  photo: PropTypes.object.isRequired,
  rating: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
};

export default RecipeHeader;
