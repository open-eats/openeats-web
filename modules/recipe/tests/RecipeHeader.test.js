import React from 'react';
import RecipeHeader from '../components/RecipeHeader';
import createComponentWithIntl from '../../../config/jest/createComponentWithIntl';

test('Test Header', () => {
  const component = createComponentWithIntl(
    <RecipeHeader
        photo={ 1 }
        title={ 'Tasty Chili' }
        rating={ 3 }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Test Header w/ no Photo', () => {
  const component = createComponentWithIntl(
    <RecipeHeader
        title={ 'Tasty Chili' }
        rating={ 3 }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Test Header w/ bad rating', () => {
  const component = createComponentWithIntl(
    <RecipeHeader
        title={ 'Tasty Chili' }
        rating={ -1 }
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
