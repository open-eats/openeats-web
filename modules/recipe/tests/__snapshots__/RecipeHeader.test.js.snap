// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header 1`] = `
<div
  className="panel-heading hero-image"
  style={
    Object {
      "backgroundImage": "url(1)",
    }
  }
>
  <div
    className="row title"
  >
    <div
      className="col-xs-12"
    >
      <h3>
        Tasty Chili
      </h3>
      <p
        className="rating-stars"
      >
        <span
          className="glyphicon glyphicon-star"
        />
        <span
          className="glyphicon glyphicon-star"
        />
        <span
          className="glyphicon glyphicon-star"
        />
        <span
          className="glyphicon glyphicon-star-empty"
        />
        <span
          className="glyphicon glyphicon-star-empty"
        />
      </p>
    </div>
  </div>
  <div
    className="row options print-hidden"
  >
    <div
      className="col-xs-12"
    >
      <button
        className="btn btn-primary"
        onClick={undefined}
      >
        <span
          aria-hidden="true"
          className="glyphicon glyphicon-calendar"
        />
      </button>
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        <span
          aria-hidden="true"
          className="glyphicon glyphicon-print"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Test Header w/ bad rating 1`] = `
<div
  className="panel-heading"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <h3>
        Tasty Chili
      </h3>
      <p
        className="rating-stars"
      >
        <span
          className="glyphicon glyphicon-star-empty"
        />
        <span
          className="glyphicon glyphicon-star-empty"
        />
        <span
          className="glyphicon glyphicon-star-empty"
        />
        <span
          className="glyphicon glyphicon-star-empty"
        />
        <span
          className="glyphicon glyphicon-star-empty"
        />
      </p>
    </div>
  </div>
</div>
`;

exports[`Test Header w/ no Photo 1`] = `
<div
  className="panel-heading"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <h3>
        Tasty Chili
      </h3>
      <p
        className="rating-stars"
      >
        <span
          className="glyphicon glyphicon-star"
        />
        <span
          className="glyphicon glyphicon-star"
        />
        <span
          className="glyphicon glyphicon-star"
        />
        <span
          className="glyphicon glyphicon-star-empty"
        />
        <span
          className="glyphicon glyphicon-star-empty"
        />
      </p>
    </div>
  </div>
</div>
`;
